example/linux/gw-example/main.c

```diff
@@ -74,23 +83,43 @@ static bool MQTT_publish(char * topic, uint8_t * payload, size_t payload_size, b
 
     if (!m_pub_queue_empty && (m_pub_queue_write == m_pub_queue_read))
     {
+        LOGE("Unable to publish, message queue full pos %d\n", m_pub_queue_write);
         ret = false;
     }
     else
     {
         // Insert our publish
         message_p = &m_publish_queue[m_pub_queue_write];
-        // TODO add check on size
-        strcpy(message_p->topic, topic);
-        memcpy(message_p->payload, payload, payload_size);
-        message_p->payload_size = payload_size;
-        message_p->retained = retained;
+        if(message_p->payload_p != NULL)
```

Comment:
Have you seen the case or is it just in case?

Reply:
Keep it just in case, before using pointer